### PR TITLE
Minor formatting and style definitions

### PR DESCRIPTION
**Pull request by wforney**
- Sealed the attribute
- Separate file per class
- Whitespace handling set to tab and let indent/tab size float (allows user to define tab width in their editor)
- Added editorconfig settings for most things, except sort order, kept preferences from existing project... except set warning on using outside namespace, but didn't move them inside

After this I will pull in the resource file changes for localization.

### CODE REVIEW

**Comment by wforney** [comment]
- Moved all strings to use resources
- Switched setting to put usings outside namespace to match previous code

**Comment by wforney** [comment]
- Added cancellation tokens to all repository async methods
- Added null argument checks

**Review by IEvangelist — CHANGES_REQUESTED** [review]
Hi @wforney - Is there any way you could give a pull request with just a few of these changes? For example, `sealed`, new file for class, new `Constants` class, `Resources`, and then that's it? Then from there we'll look again at what is needed - I have already been working on some of the other parts, like `CancellationToken`, etc.

**Comment by IEvangelist** [comment]
Addressed a lot of this in https://github.com/IEvangelist/azure-cosmos-dotnet-repository/commit/1c944b7c3b9ceecfd66204a3208e5309a4884215 and https://github.com/IEvangelist/azure-cosmos-dotnet-repository/commit/227c05d00fa7a7e03487e05417c70a8104a8f133. Thanks for these contributions.

File: Microsoft.Azure.CosmosRepository/src/Attributes/PartitionKeyPathAttribute.cs [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
-    /// By default, "/id" is used.
-    /// </remarks>
-    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
-    public class PartitionKeyPathAttribute : Attribute
-    {
-        /// <summary>
-        /// Gets or sets the path of the parition key.
-        /// </summary>
-        public string Path { get; } = "/id";
+	/// <summary>
```
[the remark this answers is not in the document]
Comment (wforney) [reply Microsoft.Azure.CosmosRepository/src/Attributes/PartitionKeyPathAttribute.cs:10]: If you use tabs you can set the tab width to 4 spaces in your editor. Other people can change it to what they want. You just force them to your preference by forcing spaces.

File: Microsoft.Azure.CosmosRepository/src/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright © IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.CosmosRepository
+{
+	/// <summary>
+	/// The constants class.
+	/// </summary>
+	public static class Constants
```
[the remark this answers is not in the document]
Comment (wforney) [reply Microsoft.Azure.CosmosRepository/src/Constants.cs:9]: No particular reason.

**Comment by wforney** [comment]
Cool, I hadn't had time during the week to mess with this.

**Pull request closed without merging**